### PR TITLE
Ensure Controller#transitionToRoute and Route#intermediateTransitionTo work in Engines

### PR DESCRIPTION
**Pull request by trentmwillis**
Addressing https://github.com/dgeb/ember-engines/issues/159 and https://github.com/dgeb/ember-engines/issues/186.

Planning to add tests (though there don't seem to be any tests for related features).

### CODE REVIEW

File: packages/ember-routing/lib/utils.js [hunk 1]
```diff
@@ -165,3 +165,26 @@ function accumulateQueryParamDescriptors(_desc, accum) {
...
+*/
+export function prefixRouteNameArg(route, args) {
+  let routeName = args[0];
+  let owner = getOwner(route);
+  let prefix = owner.mountPoint;
+
+  // only alter the routeName if it's actually referencing a route.
+  if (owner.routable && typeof routeName === 'string') {
+    if (resemblesURL(routeName)) {
+      throw new EmberError('Route#transitionTo cannot be used for URLs. Please use the route name instead.');
```
Comment (nathanhammond) [inline packages/ember-routing/lib/utils.js]: Hard-coded source `Route#transitionTo`.

**Comment by trentmwillis** [comment]
@dgeb / @rwjblue fixed prior issues and added some basic unit tests to verify programmatic transition behavior in Engines.

**Comment by rwjblue** [comment]
Oops, forgot to ask about commit prefix. I think this should target release, c/d?

**Comment by trentmwillis** [comment]
> I think this should target release, c/d?

I believe so since it's a gap in functionality of the current release.

**Comment by rwjblue** [comment]
OK, I've cherry-picked into beta, but I think we need to PR directly against release. Lots of conflicts (due to imports changing from absolute to relative and removing the engines feature flag), and I'd like to get a CI run.

Would you mind doing that?

**Comment by trentmwillis** [comment]
@rwjblue sure, I'll try to get to it sometime this weekend.

**Pull request merged**